master/internal/rm/kubernetesrm/resource_pool.go

```diff
@@ -423,11 +430,32 @@ func (k *kubernetesResourcePool) moveJob(
 	return nil
 }
 
+func (k *kubernetesResourcePool) correctJobQInfo(
+	reqs []*sproto.AllocateRequest,
+	q map[model.JobID]*sproto.RMJobInfo,
+) map[model.JobID]*sproto.RMJobInfo {
+	jobIDToPods := map[model.JobID]int{}
+	for ref, pods := range k.allocRefToRunningPods {
+		allocReq, ok := k.reqList.TaskByHandler(ref)
+		if ok {
+			jobIDToPods[allocReq.JobID] += pods
+		}
+	}
+	for id := range q {
+		if q[id].RequestedSlots <= k.config.MaxSlotsPerPod {
```

Comment:
this calculation needs to be in the ```for ref, pods := range k.allocRefToRunningPods {``` loop
 
Suppose ```maxSlotsPerPod = 2``` and two nodes with two GPUs each and we run an experiment with 4 single slot trials. RequestedSlots would be 4 and MaxSlotsPerPod = 2, giving us 8 for AllocatedSlots when we would want 4.

I think we want the first loop to look something like

```
jobIDToAllocatedSlots := map[model.JobID]int{}
for _, req := range reqs {
    if req.SlotsNeeded >= k.config.MaxSlotsPerPod {
        jobIDToAllocatedSlots[req.JobID] += allocRefToRunningPods[req.AllocationHandler] * k.config.MaxSlotsPerPod
    } else {
        jobIDToAllocatedSlots[req.JobID] += allocRefToRunningPods[req.AllocationHandler] * req.SlotsNeeded    
    }
}
```

and the second loop to use the map

Reply:
Good point -- In that implementation, I even called `reqs` but didn't actually use it in my function oops
See new changes.